src/request.rs

```diff
@@ -271,13 +297,13 @@ impl<State> Request<State> {
     /// # Panics
     ///
     /// Panic if `key` is not a parameter for the route.
-    pub fn param<T: FromStr>(&self, key: &str) -> Result<T, T::Err> {
+    pub fn param<T: FromStr>(&self, key: &str) -> Result<T, ParamError<T>> {
```

Comment:
Does it work to make ParamError generic over `T::Err`, like `Result<T, ParamError<T::Err>>` and then simplify the bounds on `ParamError` to refer directly to the contained error?

Reply:
Not AFAICT, but will try to trim down the bounds, looks a bit messy this way.

Reply:
something like this is what i had in mind:
```diff
diff --git a/src/request.rs b/src/request.rs
index fe8e16c..7950a28 100644
--- a/src/request.rs
+++ b/src/request.rs
@@ -27,15 +27,14 @@ pub struct Request<State> {
 }
 
 #[derive(Debug)]
-pub enum ParamError<T: FromStr> {
+pub enum ParamError<T> {
     NotFound(String),
-    ParsingError(T::Err),
+    ParsingError(T),
 }
 
 impl<T> std::fmt::Display for ParamError<T>
 where
-    T: FromStr + std::fmt::Debug,
-    <T as std::str::FromStr>::Err: std::fmt::Debug + std::fmt::Display,
+    T: std::fmt::Debug + std::fmt::Display,
 {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
@@ -45,12 +44,7 @@ where
     }
 }
 
-impl<T> std::error::Error for ParamError<T>
-where
-    T: FromStr + std::fmt::Debug,
-    <T as std::str::FromStr>::Err: std::fmt::Debug + std::fmt::Display,
-{
-}
+impl<T> std::error::Error for ParamError<T> where T: std::fmt::Debug + std::fmt::Display {}
 
 impl<State> Request<State> {
     /// Create a new `Request`.
@@ -297,7 +291,7 @@ impl<State> Request<State> {
     /// # Panics
     ///
     /// Panic if `key` is not a parameter for the route.
-    pub fn param<T: FromStr>(&self, key: &str) -> Result<T, ParamError<T>> {
+    pub fn param<T: FromStr>(&self, key: &str) -> Result<T, ParamError<T::Err>> {
         self.route_params
             .iter()
             .rev()
```

Reply:
Great! Thanks for the quick reply!!

Reply:
... and thanks for filling in for my dead brain cells :) it's 4am, need to go to bed.
Thanks again for the help and quick responses!